'sun' is considered a forbidden identifier by GCC and therefore it creates
a "fixed" version of bash/include/memalloc.h in its library directory.

This prevents the compiler from seeing any updated version of that file.

To resolve that, the fix that GCC would use is applied directly to the
include file so that it doesn't see a need to fix it.

diff -wpruN '--exclude=*.orig' a~/include/memalloc.h a/include/memalloc.h
--- a~/include/memalloc.h	1970-01-01 00:00:00
+++ a/include/memalloc.h	1970-01-01 00:00:00
@@ -22,7 +22,7 @@
 #if !defined (_MEMALLOC_H_)
 #  define _MEMALLOC_H_
 
-#if defined (sparc) && defined (sun) && !defined (HAVE_ALLOCA_H)
+#if defined (sparc) && defined (__sun__) && !defined (HAVE_ALLOCA_H)
 #  define HAVE_ALLOCA_H
 #endif
 
